flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingMetricCollector.java

```diff
@@ -147,7 +149,13 @@ public CollectedMetricHistory updateMetrics(
         if (now.isBefore(windowFullTime)) {
             LOG.info("Metric window not full until {}", readable(windowFullTime));
         } else {
-            collectedMetrics.setFullyCollected(true);
+            if (isExcluded) {
+                LOG.info(
+                        "Autoscaling on halt based on exclusion rule {}",
+                        conf.get(AutoScalerOptions.EXCLUDED_PERIODS));
+            } else {
+                collectedMetrics.setFullyCollected(true);
+            }
```

Comment:
Enforcing this through collectedMetrics.setFullyCollected(true); seems a bit hacky and susceptible to breaking if we treat the collected metrics differently.

Maybe we should move this check to the `ScalingExecutor`

Reply:
what do you think @mxm ?

Reply:
Although the current logic works fine for blocking the scaling during the configured phases, I agree that the logic would be better suited for ScalingExecutor. There are other features, e.g. recommended parallelism, which will stop working if the metric collection phase hasn't completed. We only want to block executing the scaling. All other logic should run normally.

Reply:
@gyfora @mxm you are right, it's better to place the excludedPeriods blocking logic in ScalingExecutor. As the meaning of excludedPeriods is very similar with config `scaling.enabled`, i combines the logic in `ScalingExecutor`. Now excludedPeriods blocking works well with recommended parallelism